geanyprj/src/project.c

```diff
@@ -311,6 +315,14 @@ void geany_project_set_run_cmd(struct GeanyPrj *prj, const gchar *run_cmd)
 }
 
 
+void geany_project_set_working_dir(struct GeanyPrj *prj, const gchar *working_dir)
+{
+	if (prj->working_dir)
+		g_free(prj->working_dir);
```

Comment:
(minor) You can `g_free` unconditionally since freeing NULL is a valid no-op (cd manual [here](https://developer.gnome.org/glib/stable/glib-Memory-Allocation.html#g-free)).